src/Parsing/String.purs

```diff
@@ -249,9 +262,11 @@ match p = do
 -- | regex "x*" (dotAll <> ignoreCase)
 -- | ```
 -- |
--- | The `dotAll`, `unicode`, and `ignoreCase` flags might make sense for
--- | a `regex` parser. The other flags will
--- | probably cause surprising behavior and you should avoid them.
+-- | The `dotAll`, `multiline`, `unicode`, and `ignoreCase` flags might make
+-- | sense for a `regex` parser. In fact, per JS RegExp semantics matching a
```

Comment:
I don't think I want to encourage people to use the `multiline` flag until we have added a bunch of `multiline` test cases to the test suite to be sure that `multiline` parsing works the way that we think it does.

Reply:
Well, that makes whole PR a moot point. `multiline` is the behavior people users would expect from regular expressions *(as explained in the commit message)*, which is why after stumbling upon odd behavior in the parser and finding out that the usual behavior of regex isn't the one that `noFlags` give, I went to document how it should work. Now you're pushing against. I don't get that, but you're the maintainer, so ok.

Reply:
> `multiline` is the behavior users would expect from regular expressions

to add to that: besides the explanation referred, there's also a point that explicitly matching newline characters is often frowned upon. This is because there exist at least 3 different styles of newlines. So the usual advice is not to match a `\n` for example, but to match a `$` instead.

Reply:
I think personally I just view this parser as serving a different purpose than for what enabling `multiline` would give it. I view it as similar to `satisfy`, it's just there to define a range or pattern of acceptable characters. If I wanted to be able to skip ahead over an arbitrary number of lines before finding something I'd be doing that explicitly using other parser combinators.

It's already compromised regex interface too - you can't really use it the way you normally would since the groups in the pattern are inaccessible.

(I'm not saying I'm right in this opinion, just giving context for why I wouldn't miss `multiline` at all).

Reply:
> I think personally I just view this parser as serving a different purpose than for what enabling `multiline` would give it. I view it as similar to `satisfy`, it's just there to define a range or pattern of acceptable characters. If I wanted to be able to skip ahead over an arbitrary number of lines before finding something I'd be doing that explicitly using other parser combinators.

FWIW, the current documentation encourages to prefer `regex` over other parsers. Quoting the relevant block:

> This parser may be useful for quickly consuming a large section of the input `String`, because in a JavaScript runtime environment a compiled RegExp is a lot faster than a monadic parser built from parsing primitives.

--------

> It's already a compromised regex interface too - you can't really use it the way you normally would since the groups in the pattern are inaccessible.

Fair, although till now I haven't even noticed lack of groups, because the "combinators style" implies that when there's a pattern one wants to retrieve, you'd just consume it separately.

Reply:
> Well, that makes whole PR a moot point. `multiline` is the behavior users would expect from regular expressions, which is why after stumbling upon odd behavior in the parser and finding out that the usual behavior of regex isn't the one that `noFlags` gives, I went to document how it should work.

Do we even understand how it works though? I checked @garyb ’s case and confirmed that

```haskell
  runParser "some\nvarious\nlines" do
    m <- fromRight' unsafeCoerce $ regex "various$" multiline
    r <- rest
    pure $ Tuple m r
```
gives the result
```haskell
(Right (Tuple "various" "rious\nlines"))
```

which is surprising and wrong.

So I think that the current advice in the docs

![image](https://github.com/user-attachments/assets/bfbffa55-7e1e-4710-b6a6-4505dc842a19)

is pretty good advice until we decide what kind of behavior the `multiline` flag should have, fix it so that it behaves that way, and test it.

Reply:
Thanks for calling attention to this, though. It would be nice if the `multiline` flag just worked in an intuitively correct way, but it currently doesn't.